VR-8152 - load project visibility from uac

coryverta opened the pull request:
Refactor the population of the project visibility in `getProjectById` to load from UAC.

coryverta committed:
Implement call to setResources

coryverta committed:
Set ownerID to be the parsed Long user ID

coryverta committed:
Merge branch 'master' into cj/VR-8152/project-creation-uses-setResources

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/authservice/AuthServiceChannel.java

coryverta committed:
Replace legacy call to create workspace role bindings

Use the new path to UAC instead.

coryverta committed:
Merge branch 'master' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
WIP

coryverta committed:
Rename createWorkspaceRoleBindings -> createWorkspacePermissions

coryverta committed:
Introduce createWorkspacePermissions override that accepts the name

coryverta committed:
getResourceVisibility support org-private

coryverta committed:
Allow ownerId to be optional

coryverta committed:
Allow projectType to be optional

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge remote-tracking branch 'origin/cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Merge proto changes

coryverta committed:
Migrate to VisibilityEnum.Visibility

coryverta committed:
Test fixes

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java
#	backend/src/main/java/ai/verta/modeldb/entities/ProjectEntity.java
#	backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

coryverta committed:
WIP

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Remove Project.SetProjectVisibility

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

coryverta committed:
Add RoleService.getResources

conradoverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -1348,10 +1299,18 @@ public Project setProjectWorkspace(String projectId, WorkspaceDTO workspaceDTO)
+          projectEntity.getId(),
+          ownerId,
+          ModelDBServiceResourceTypes.PROJECT,
+          CollaboratorTypeEnum.CollaboratorType.READ_WRITE,
```
Why did we change this?

anandjakhaniya committed:
Remove unused code for the creating RoleBinding while insert project on mdb

anandjakhaniya committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' of github.com:VertaAI/modeldb into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Implement visibility fetching from UAC in getProjectById

coryverta committed:
Implement visibility fetching from UAC in getProjectById

coryverta committed:
Add utility method to fetch project visibility.

Migrated getProjectById to use new method.

coryverta committed:
getProjectVisibility requires the workspace type

coryverta commented on backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java:
```diff
@@ -1150,6 +1150,41 @@ public WorkspaceDTO getWorkspaceDTOByWorkspaceId(
+      switch(resourceVisibility) {
+        case PRIVATE:
+          return VisibilityEnum.Visibility.PRIVATE;
+        case ORG_DEFAULT:
```
@conradoverta what should this be?  I can't figure out how to convert this.

conradoverta replied:
Why are we doing the conversion from the resource visibility to the project visibility?

coryverta replied:
UAC returns a ResourceVisibility, but the Project has a VisibilityEnum.Visibility

conradoverta replied:
I'd recommend changing the project representation for full compatibility. Otherwise we won't be able to represent the full range of possibilities. And yes, I'm aware this is a breaking change. Need to check the impact on the client.

coryverta committed:
Code reformat

anandjakhaniya committed:
revert old project visibility from ProjectEntity.java for migration use

anandjakhaniya committed:
populate visibility from UAC to the fetched projects

anandjakhaniya committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

anandjakhaniya committed:
Fix getResource service on RoleServiceUtils.java

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' of github.com:VertaAI/modeldb into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

coryverta committed:
Support ORG_DEFAULT visibility for Project

coryverta committed:
Populate project visibility in getProjectEntityByBatchIds

coryverta committed:
Utilize CollaboratorService.DeleteResources to clean up on project deletion

Add new utility methods for invoke DeleteResources and for cleaning up project resources

anandjakhaniya committed:
Fix setProjectWorkspace endpoint (#1758)

Remove setVisibility DAO impl

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

coryverta committed:
Utilize CollaboratorService.DeleteResources to clean up on project deletion
Re-add import lost in merge

coryverta committed:
Cleanup unused methods

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Remove ProjectService.setProjectWorkspace

coryverta committed:
Remove ProjectService.setProjectWorkspace

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

coryverta committed:
Code formatter

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/99

coryverta committed:
Merge branch 'cj/VR-8152/use-collaborator-visibility' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Migrate to using ResourceVisibility for ProjectEntity

coryverta committed:
Migrate to using ResourceVisibility for ProjectEntity

coryverta committed:
Merge branch 'cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/load-project-visibility-from-uac

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java

coryverta committed:
Migrate to using ResourceVisibility for ProjectEntity

coryverta committed:
Migrate to using ResourceVisibility for ProjectEntity

coryverta committed:
Merge branch 'master' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Remove support for PUBLIC visibility in RoleService.getReadOnlyAccessibleResourceIds

conradoverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -750,6 +724,8 @@ public Project deepCopyProjectForUser(String srcProjectID, UserInfo newOwner)
       transaction.commit();
       LOGGER.debug("Project deleted successfully");
+
+      roleService.deleteProjectResources(allowedProjectIds);
```
We should let the GC process handle this during soft delete.

coryverta replied:
+:+1:

coryverta committed:
Update backend/src/main/java/ai/verta/modeldb/project/ProjectDAO.java

Co-authored-by: conradoverta <53195685+conradoverta@users.noreply.github.com>

conradoverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -902,7 +879,19 @@ public Project setProjectShortName(String projectId, String projectShortName, Us
+                      projectEntity.getId(),
+                      projectEntity.getWorkspace(),
+                      projectEntity.getWorkspace_type());
+              projectEntity.setProjectVisibility(projectVisibility);
```
We need to set workspace and owner too, right?

coryverta replied:
Yes, I will need to change this to fix that.  I am currently relying upon the DB values in MDB, but those aren't the correct source of truth now.

conradoverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -902,7 +879,19 @@ public Project setProjectShortName(String projectId, String projectShortName, Us
     query.setParameterList("ids", projectIds);
-    return query.list();
+    final List<ProjectEntity> projectEntities = query.list();
+    return projectEntities.stream()
```
```suggestion
    // TODO: remove UAC fields from the MDB protobuf representation?
    return projectEntities.stream()
```

conradoverta commented on backend/src/main/java/ai/verta/modeldb/entities/ProjectEntity.java:
```diff
@@ -347,6 +348,12 @@ public Project getProtoObject() throws InvalidProtocolBufferException {
 
+    WorkspaceDTO workspaceDTO =
+        roleService.getWorkspaceDTOByWorkspaceId(null, this.workspace, this.workspace_type);
+    projectBuilder.setVisibility(
```
Need to set owner and workspace too.

coryverta committed:
Update backend/src/main/java/ai/verta/modeldb/utils/RdbmsUtils.java

Co-authored-by: conradoverta <53195685+conradoverta@users.noreply.github.com>

coryverta committed:
Update backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java

Co-authored-by: conradoverta <53195685+conradoverta@users.noreply.github.com>

conradoverta commented on backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java:
```diff
@@ -1150,6 +1153,150 @@ public WorkspaceDTO getWorkspaceDTOByWorkspaceId(
+    return ResourceVisibility.PRIVATE;
+  }
+
+  private ResourceVisibility getResourceVisibility(
```
I don't see this being used where. Could you clarify where it's used? I mainly want to understand using that visibility enum.

coryverta replied:
It's not used, removing it.

conradoverta commented on backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java:
```diff
@@ -1150,6 +1153,150 @@ public WorkspaceDTO getWorkspaceDTOByWorkspaceId(
     }
   }
 
+  private VisibilityEnum.Visibility fromResourceVisibility(
```
Where is this used?

coryverta replied:
It's not used, removing it.

coryverta committed:
Merge conflicts

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/7/

coryverta committed:
Code formatter

coryverta committed:
Merge remote-tracking branch 'origin/cj/VR-8152/project-creation-uses-setResources' into cj/VR-8152/project-creation-uses-setResources

coryverta committed:
Inline the call to createWorkspacePermissions in insertProject

Default the ownerId to empty so UAC populates it.
Set the default collab type to READ_ONLY

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/10/

coryverta committed:
Merge

coryverta committed:
Update backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

Co-authored-by: conradoverta <53195685+conradoverta@users.noreply.github.com>

coryverta committed:
Remove unused methods

coryverta committed:
Remove explicit call to delete resources during project deletion.  This is handled by the cron task.

coryverta committed:
Refactor RoleService to return the full resource for the project instead of just ot visibility.

This allows the caller to fetch the visibility, owner, and workspace ID all in a single call.

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/17/

coryverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java:
```diff
@@ -1330,47 +1273,6 @@ public void getProjectShortName(
     }
   }
 
-  @Override
```
Removed from protos

coryverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java:
```diff
@@ -1696,47 +1598,4 @@ public void deleteProjects(
   }
-
-  @Override
-  public void setProjectWorkspace(
```
Removed from protos

coryverta commented on backend/src/test/java/ai/verta/modeldb/FindProjectEntitiesTest.java:
```diff
@@ -1166,70 +1167,6 @@ public void findAndSortProjectsByAttributeTest() {
 
-  /** Find public projects sort by name */
-  @Test
-  public void findPublicProjectsSortingByNameTest() {
```
Removed support fo PUBLIC projects

coryverta commented on backend/src/test/java/ai/verta/modeldb/HydratedServiceTest.java:
```diff
@@ -2868,34 +2856,6 @@ public void findHydratedProjectsTest() {
       }
     }
 
-    keyValueQuery =
```
Removed support for PUBLIC projects

coryverta commented on backend/src/test/java/ai/verta/modeldb/ProjectTest.java:
```diff
@@ -2033,67 +2021,6 @@ public void nnnn_setEmptyProjectReadMeTest() throws IOException {
     LOGGER.info("Get Project ReadMe test stop................................");
   }
 
-  @Test
```
Removed proto

conradoverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -1405,7 +1203,7 @@ public Project setProjectWorkspace(String projectId, WorkspaceDTO workspaceDTO)
           roleService.getAccessibleResourceIds(
```
Note: I see no reason for this to require visibility.

conradoverta replied:
We should follow up on this later.

coryverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -747,24 +687,8 @@ public Project deepCopyProjectForUser(String srcProjectID, UserInfo newOwner)
     return newProject;
   }
 
-  private List<String> getWorkspaceRoleBindings(
```
Unused code

coryverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -922,111 +847,13 @@ public Project setProjectShortName(String projectId, String projectShortName, Us
   }
 
-  @Override
-  public Project setProjectVisibility(String projectId, ProjectVisibility projectVisibility)
```
Unused code

coryverta commented on backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:
```diff
@@ -1351,41 +1184,6 @@ public Project deleteArtifacts(String projectId, String artifactKey)
   }
 
-  @Override
-  public Project setProjectWorkspace(String projectId, WorkspaceDTO workspaceDTO)
```
Unused code

coryverta committed:
Fix merge conflicts

coryverta committed:
Merge remote-tracking branch 'origin/cj/VR-8152/load-project-visibility-from-uac' into cj/VR-8152/load-project-visibility-from-uac

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/21/
